controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -252,7 +252,7 @@ func (r *StorageClassClaimReconciler) reconcileConsumerPhases() (reconcile.Resul
 				r.log.Error(fmt.Errorf("storageClassClaim is not compatible with existing StorageClass"),
 					"StorageClassClaim validation failed.")
 				r.storageClassClaim.Status.Phase = v1alpha1.StorageClassClaimFailed
-				return reconcile.Result{}, nil
+				return reconcile.Result{}, fmt.Errorf("failed to validate storageclassclaim, %t %s %s", scHasEncryptionMethod, sccEncryptionMethod, scEncryptionMethod)
```

Comment:
The reason maybe a mismatch in fs/block types too ?

Reply:
let's add error logs, right next to the lines where the invalid condition is met.